src/Util/Emailer.php

```diff
@@ -161,12 +161,17 @@ public function send(IEmail $email)
 		                . rand(100000000, 999999999) . '=:'
 		                . rand(10000, 99999);
 
+		$messageHeader = $email->getAdditionalMailHeaderString();
+		if ($countMessageId === 0) {
+			$messageHeader .= 'Message-ID: <Friendica-Util-Emailer-' . Strings::getRandomHex() . '@' . $this->baseUrl->getHost() . '>' . "\r\n";
```

Comment:
Isn't there any already exiting function to create a message id?

Reply:
Nope. We're setting the `Message-ID` in two places, in the `mailstream` addon and in the Notify repository. Both are using a different method, one base on a URI and the second based on a GUID. Both are using the server hostname (with `gethostname()`).